neqo-transport/src/send_stream.rs

```diff
@@ -1232,4 +1257,35 @@ mod tests {
         assert!(matches!(&f4_token, Some(RecoveryToken::Stream(x)) if x.length == 10));
         assert!(matches!(&f4_token, Some(RecoveryToken::Stream(x)) if x.fin));
     }
+
+    #[test]
+    fn send_atomic() {
+        let flow_mgr = Rc::new(RefCell::new(FlowMgr::default()));
+        flow_mgr.borrow_mut().conn_increase_max_credit(5);
+        let conn_events = ConnectionEvents::default();
+
+        let mut s = SendStream::new(4.into(), 0, Rc::clone(&flow_mgr), conn_events);
+        s.set_max_stream_data(2);
+
+        // Stream is initially blocked (conn:5, stream:2)
+        // and will not accept atomic write of 3 bytes.
+        assert_eq!(s.send_atomic(b"abc").unwrap(), 0);
+
+        // assert non-atomic write works
+        assert_eq!(s.send(b"abc").unwrap(), 2);
```

Comment:
This should still send a blocked frame.  This test doesn't appear to test that these arrive at the right moments.  Can you do that?